http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java

```diff
@@ -877,104 +906,82 @@ public void testDowngradedHttpProtocolUsed()
     }
 
     @Test
-    public void testPutMethodWithLargeStreamingBodyGenerator()
+    public void testPipedLargeContent()
             throws Exception
     {
-        testPutMethodWithStreamingBodyGenerator(true);
+        testPiped(true);
     }
 
     @Test
-    public void testPutMethodWithStreamingBodyGenerator()
+    public void testPipedSmallContent()
             throws Exception
     {
-        testPutMethodWithStreamingBodyGenerator(false);
+        testPiped(false);
     }
 
-    protected void testPutMethodWithStreamingBodyGenerator(boolean largeContent)
+    protected void testPiped(boolean largeContent)
             throws Exception
     {
         String content = largeContent ? LARGE_CONTENT : "hello".repeat(1000);
 
-        try (CloseableTestHttpServer server = newServer()) {
-            URI uri = server.baseURI().resolve("/road/to/nowhere");
-            server.servlet().setResponseBody(content);
+        var locals = new Object()
+        {
+            String putContent;
+            CloseableTestHttpServer server;
+        };
 
-            ResponseHandler<Void, RuntimeException> responseHandler = new ResponseHandler<>()
+        try (JettyHttpClient httpClient = new JettyHttpClient("streaming-test", createClientConfig())) {
+            HttpServlet pipeServlet = new HttpServlet()
             {
                 @Override
-                public Void handleException(Request request, Exception exception)
-                        throws RuntimeException
+                protected void service(HttpServletRequest request, HttpServletResponse response)
+                        throws IOException
                 {
-                    throw propagate(request, exception);
-                }
+                    switch (request.getPathInfo()) {
+                        case "/get" -> {
+                            response.setIntHeader("Content-Length", content.length());
+                            response.setStatus(200);
+                            ByteStreams.copy(new ByteArrayInputStream(content.getBytes(UTF_8)), response.getOutputStream());
```

Comment:
Can use `transferTo()`
```suggestion
                            new ByteArrayInputStream(content.getBytes(UTF_8)).transferTo(response.getOutputStream());
```

Reply:
Actually, why not do
```suggestion
                            response.getOutputStream().write(content.getBytes(UTF_8));
```